### PR TITLE
ham10000 classification  data module

### PR DESCRIPTION
**Pull request by HAikalfebrianp96**
# Description

Make small modifications to the data module, add a lightning module for breast cancer classification using ResNet-18, and training pipeline with a single train, validation, and test step for a fast development run

- New Features:

1. modification to the data module
2. Adding ResNet-18 architecture using torch module
3. Create a lightning module for skin cancer classification using ResNet-18
4. Create Hydra config for data module + model
5. Create testing functions for the training pipeline

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran tests using pytest to ensure the correctness of the changes.
- [x]  Tested with a dummy dataset
- [x] Tested the modified data module.
- [x] Tested the training pipeline

**Test Configuration**:

- Firmware version:
- Hardware: lenovo  Intel(R) Core(TM) i5-7200U CPU @  2.50 GHz
- Toolchain: Visual Studio Code (VSCode)
- SDK: Windows 10

# Checklist:

- [x] Did you make sure title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [x]  I have commented my code, particularly in hard-to-understand areas
- [x]  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by haritsahm** [comment]
Please use the following PR [template](https://github.com/KalbeDigitalLab/AI-Vision-Lightning-Pipeline/blob/main/.github/PULL_REQUEST_TEMPLATE.md)

File: src/data/ham1000_datamodule.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+from typing import Optional, Tuple, Union
+
+import deeplake
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, Dataset 
+from torchvision import transforms
+from src.data.components.ham1000_parser import ham1000dataset
+
+class HamCancerDataModule(LightningDataModule):
+    def __init__(
```
Comment (haritsahm) [inline src/data/ham1000_datamodule.py:10]: Missing docstring, use the numpy doc style

File: src/data/components/ham1000_parser.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import numpy as np
+import torch
+import torchvision
+from torch.utils.data import Dataset
+
+
+class ham1000dataset(Dataset):
+    """DeepLakeDataset. A PyTorch Dataset Wrapper for DeepLake Datasets.
+
+    Parameters:
```
Comment (haritsahm) [inline src/data/components/ham1000_parser.py:12]: Incorrect type docstring style, please use the numpy doc style

**Review by haritsahm — CHANGES_REQUESTED** [review]
Please read the contribution guideline

**Comment by HAikalfebrianp96** [comment]
> Please use the following PR [template](https://github.com/KalbeDigitalLab/AI-Vision-Lightning-Pipeline/blob/main/.github/PULL_REQUEST_TEMPLATE.md)

**Pull request closed without merging**